src/main/java/org/opensearch/security/opensaml/integration/CleanerCreator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.opensaml.integration;
+
+import org.opensearch.common.util.concurrent.OpenSearchExecutors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.ref.Cleaner;
+
+public class CleanerCreator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CleanerCreator.class);
+
+    /** Constructor. */
+    private CleanerCreator() {}
+
+    public static Cleaner create(final Class<?> requester) {
+        // Current approach here is to create a new Cleaner on each call. A given class requester/owner
+        // is assumed to call only once and store in static storage.
+        LOG.debug("Creating new java.lang.ref.Cleaner instance requested by class:  {}", requester.getName());
+        return Cleaner.create(OpenSearchExecutors.daemonThreadFactory("cleaners"));
```

Comment:
We probably need to create daemon thread factory only once and not for each cleaner

Reply:
> on other side, if so, why should a threadFactory be opened for each cleaner?

This is how the `OpenSearch` manages permissions by creating the dedicated thread factories